lib/generate_letters.ml

```diff
@@ -38,5 +38,104 @@ let rec index_of_letter curr letter = function
   | h :: t ->
       if h = letter then curr else index_of_letter (curr + 1) letter t
 
-(** used http://www.breakingthegame.net/leaves2 to generate accurate
-    probability rates for vowels and consonants*)
+(* vowels is a list containing every vowel char accepted by our Scrabble
+   game*)
+let vowels = [ 'A'; 'E'; 'I'; 'O'; 'U' ]
+
+(* consonants is a list containing every consonant char accepted by our
+   Scrabble game *)
+let consonants =
+  [
+    'B';
+    'C';
+    'D';
+    'F';
+    'G';
+    'H';
+    'J';
+    'K';
+    'L';
+    'M';
+    'N';
+    'P';
+    'Q';
+    'R';
+    'S';
+    'T';
+    'V';
+    'W';
+    'X';
+    'Y';
+    'Z';
+  ]
+
+(* probability weights associated with each number of vowels in a
+   starting hand. For implementer reference.*)
+(**let weights =
+   [ (0, 0.07); (1, 0.25); (2, 0.36); (3, 0.24); (4, 0.07); (5, 0.01) ]*)
+
+(* alias table reference https://www.keithschwarz.com/darts-dice-coins/*)
+(* probability list needed for an alias table*)
+let prob = [ 0.42; 0.92; 0.56; 1.; 0.42; 0.06 ]
+
+(* alias list needed for an alias table*)
+let alias = [ 1; 2; 3; 0; 2; 2 ]
+
+(** [biased prob total] is true for heads and false for tails. This
+    method simulates a biased coin, with a [prob]/[total] chance of
+    being heads. Requires [prob] <= [total].*)
+let biased prob total =
+  let pick = Random.int total in
+  if pick < prob then true else false
+
+(** [index_item ind curr lst] is the item of lst at index ind. The
+    parameter curr is the current index.*)
+let rec index_item curr ind = function
+  | [] -> raise Not_found
+  | h :: t -> if curr = ind then h else index_item (curr + 1) ind t
+
+(** [get_rand lst] is a random element from lst. *)
+let get_rand lst =
+  let ind = Random.int (List.length lst) in
+  index_item 0 ind lst
+
+(** [num_vowels] is the number of vowels a player's starting hand based
+    on a series of probabilities.*)
+let num_vowels () =
+  let side = Random.int 6 in
+  let pick = index_item 0 side prob in
+  let h_or_t = biased (int_of_float pick * 100) 100 in
+  if h_or_t then side else index_item 0 side alias
+
+(** [lst_of size lst repo] is a list of size [size] of elements from
+    [repo]. Requires [lst] to be empty when first applied.*)
+let rec lst_of size lst repo =
```

Comment:
consider renaming to `pick_random_sample` or `pick_random_items`. also, instead of requiring that `lst` is empty, you could just have a recursive inner function which this function always calls with an empty list.